ark/type/parser/objectLiteral.ts

```diff
@@ -24,8 +28,41 @@ const stringAndSymbolicEntriesOf = (o: Record<string | symbol, unknown>) => [
 export const parseObjectLiteral = (def: Dict, ctx: ParseContext) => {
 	const required: Inner<"required">[] = []
 	const optional: Inner<"optional">[] = []
+
+	// We only allow a spread operator to be used as the first key in an object
+	// because to match JS behavior any keys before the spread are overwritten
+	// by the values in the target object, so there'd be no useful purpose in having it
+	// anywhere except for the beginning.
+	// Discussion in ArkType Discord:
+	// https://discord.com/channels/957797212103016458/1103023445035462678/1182814502471860334
+	let hasSeenFirstKey = false
+
 	for (const entry of stringAndSymbolicEntriesOf(def)) {
 		const result = parseEntry(entry)
+
+		if (result.kind === "spread" && hasSeenFirstKey) {
+			return throwParseError(
+				"Spread operator may only be used as the first key in an object"
+			)
+		} else if (result.kind === "spread") {
+			const spreadNode = ctx.scope.parse(result.innerValue, ctx)
+
+			if (
+				spreadNode.kind !== "intersection" ||
+				!spreadNode.extends(keywords.object)
+			) {
+				return throwParseError(
+					writeInvalidSpreadTypeMessage(printable(result.innerValue))
+				)
+			}
+
+			// For each key on spreadNode, add it to our object.
+			required.push(...(spreadNode.required ?? []))
```

Comment:
This is not a bad approach, but I think I'd prefer filtering the spread keys we add initially to not include those that exist as a base key.

That way we don't have to do the existing/splice stuff later.